director/director.go

```diff
@@ -388,79 +389,179 @@ func (d *Director) handleCheckConfigChange(resp *etcd.Response) error {
 	log.Debugf("%v-handleCheckConfigChange: Received new response for key %v",
```

Comment:
This is the bananafest method that needed all of the new logic for determining where a check should go. I tried my best to keep it clear -- ultimately it should probably be split up into two methods to lower the large number of conditionals.